### PR TITLE
Fix FormatString EnforcedStyle: sprintf good example

### PR DESCRIPTION
**Pull request by matass**
```
Style/FormatString: Favor format over sprintf.
      sprintf('%10s', 'hoge')
      ^^^^^^^
```

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/format_string.rb [hunk 1]
```diff
@@ -21,11 +21,11 @@ module Style
...
       # @example EnforcedStyle: sprintf
       #   # bad
-      #   puts format('%10s', 'hoge')
-      #   puts '%10s' % 'hoge'
+      #   puts sprintf('%10s', 'hoge')
       #
       #   # good
-      #   puts sprintf('%10s', 'hoge')
+      #   puts format('%10s', 'hoge')
+      #   puts '%10s' % 'hoge'
```
Comment (mikegee) [inline lib/rubocop/cop/style/format_string.rb]: Let's leave this one "bad". I get:

```
test/test.rb:1:13: C: Style/FormatString: Favor format over String#%.
puts '%10s' % 'hoge'
            ^
```

File: lib/rubocop/cop/style/format_string.rb [hunk 2]
```diff
@@ -21,19 +21,19 @@ module Style
       #
       # @example EnforcedStyle: sprintf
       #   # bad
-      #   puts format('%10s', 'hoge')
+      #   puts sprintf('%10s', 'hoge')
       #   puts '%10s' % 'hoge'
       #
       #   # good
-      #   puts sprintf('%10s', 'hoge')
+      #   puts format('%10s', 'hoge')
```
[the remark this answers is not in the document]
Comment (matass) [reply lib/rubocop/cop/style/format_string.rb]: that's interesting. I cant reproduce error with `format('%10s', 'hoge')`

[the remark this answers is not in the document]
Comment (koic) [reply lib/rubocop/cop/style/format_string.rb]: Are you customizing `EnforcedStyle` of this cop?
http://rubocop.readthedocs.io/en/latest/cops_style/#styleformatstring

The default `EnforcedStyle: format` does not register an offense using `format('%10s', 'hoge')`
https://github.com/bbatsov/rubocop/pull/5839/files#diff-ab2c55b3e5235f61b0603c43b8736548R14

**Comment by bbatsov** [comment]
It seems you misunderstood the documentation - it lists different examples for the different style supported by the cop. I don't see anything wrong with the documentation.

**Pull request closed without merging**